consensus/istanbul/validator/validator.go

```diff
@@ -42,7 +42,7 @@ func New(addr common.Address) istanbul.Validator {
 func NewValidatorSet(addrs []common.Address, proposerPolicy istanbul.ProposerPolicy, subGroupSize uint64, chain consensus.ChainReader) istanbul.ValidatorSet {
 	var valSet istanbul.ValidatorSet
 	if proposerPolicy == istanbul.WeightedRandom {
-		valSet = NewWeightedCouncil(addrs, nil, nil, nil, proposerPolicy, subGroupSize, 0, 0, chain)
+		valSet = NewWeightedCouncil(addrs, nil, nil, nil, nil, proposerPolicy, subGroupSize, 0, 0, chain)
```

Comment:
Why is `demotedAddrs` always `nil`? Does `backend.getValidators()` needs the value?

Reply:
`backend.getValidators()` calls `NewValidatorSet` when getting snapshot is failed. `NewValidatorSet` is used with proper parameters when initially creating a snapshot only.